File: src/serverless/BlazingPizza.Functions.FSharp/Specials.fs
namespace BlazingPizza.Functions.FSharp

open Microsoft.AspNetCore.Mvc
open Microsoft.Azure.WebJobs
open Microsoft.Azure.WebJobs.Extensions.Http
open Microsoft.AspNetCore.Http
open Microsoft.Extensions.Logging

open BlazingPizza

module Specials =
    [<FunctionName("Specials")>]
    let run
        ([<HttpTrigger(AuthorizationLevel.Function, "get", "post", Route = null)>] req: HttpRequest)
        ([<CosmosDB("blazingPizza",
                    "pizza",
                    ConnectionStringSetting = "CosmosConnectionString",
                    PartitionKey = "specials")>] pizzas: PizzaSpecial seq)
        (log: ILogger)
        =
        async { return OkObjectResult pizzas :> IActionResult }
        |> Async.StartAsTask

File: src/serverless/BlazingPizza.Functions.CSharp/Specials.cs
using Microsoft.AspNetCore.Mvc;
using Microsoft.Azure.WebJobs;
using Microsoft.Azure.WebJobs.Extensions.Http;
using Microsoft.AspNetCore.Http;
using Microsoft.Extensions.Logging;
using System.Collections.Generic;

namespace BlazingPizza.Functions.CSharp
{
    public static class Specials
    {
        [FunctionName("Specials")]
        public static IActionResult Run(
            [HttpTrigger(AuthorizationLevel.Function, "get", Route = null)] HttpRequest req,
            [CosmosDB("blazingPizza",
                      "pizza",
                      ConnectionStringSetting = "CosmosConnectionString",
                      SqlQuery = "SELECT * FROM c WHERE c.partitionKey = 'specials'")] IEnumerable<PizzaSpecial> pizzas,
            ILogger log)
        {
            return new OkObjectResult(pizzas);
        }
    }
}
